test/test_sparse.py

```diff
@@ -143,6 +143,79 @@ def f5():
             # Check warn-once:
             self.assertEqual(len(cm.warnings), 1)
 
+    def test_gradcheck_check_sparse_nnz(self):
+        """Tests for deprecated check_sparse_nnz keyword argument of gradcheck.
+
+        Suggested deprecation steps:
+        2.1: Specification of check_sparse_nnz triggers a warning.
+        2.2: Specification of check_sparse_nnz triggers a warning and
```

Comment:
My idea with 2.1->2.2->2.3 was that the intermediate step 2.2 will give a hint to a user on how to (unintelligently) update their code to avoid the exception. I would be fine to skip the intermediate step and go with 2.1->2.2.

@albanD could you give a resolution here?

Reply:
Why can't you just have the hint in the exception?

Reply:
When we remove `check_sparse_nnz` argument, we cannot hint at the exception because the exception will be `TypeError: "gradcheck got an unexpected keyword argument 'check_sparse_nnz"` from Python.

Reply:
Another question is whether stretching the deprecation process of a seldom-used argument is worth it.

Reply:
I would agree with Jeffrey that we can go with the simpler option here as this argument is seldom used indeed.
And if a user gets the unexpected keyword error and google `check_sparse_nnz` they should be able to find the right thing (a google search for "check_sparse_nnz pytorch" already points to this PR haha)

Reply:
lol I realize I misunderstand 2.3 in the original proposal to mean keep the argument but raise an exception. Thanks @pearu for clarifying. 

Both proposals make sense to me now. But since this flag is seldom used, fast removal sounds good.

Reply:
Fixed to a two-step deprecation process. Thank you all for the feedback!